sklearn/utils/stats.py

```diff
@@ -7,30 +7,35 @@
 
 
 def _weighted_percentile(array, sample_weight, percentile=50):
-    """Compute weighted percentile
+    """Compute the lower value(s) at a given weighted percentile.
 
-    Computes lower weighted percentile. If `array` is a 2D array, the
-    `percentile` is computed along the axis 0.
+    If `array` is a 2D array, the `values` are selected along axis 0.
 
         .. versionchanged:: 0.24
             Accepts 2D `array`.
 
+        .. versionchanged:: 1.6
+            Supports handling of `NaN` values. For `NaN` inputs, their corresponding
+            weights are set to 0 in `sample_weight`.
+            Percentiles that point to `NaN` values are redirected to the next lower
+            value if it exists.
```

Comment:
Let's skip the `..versionchanged` entry concise and move this important information in the main paragraph instead.

```suggestion
            Supports handling of `NaN` values.
```